pkg/controller/statusmanager/status_manager.go

```diff
@@ -245,10 +245,12 @@ func (status *StatusManager) GetContainers() []opv1alpha1.Container {
 // no-op if the StatusManager is currently marked as failing due to a configuration error.
 func (status *StatusManager) SetFromPods() {
 	progressing := []string{}
+	daemonSets := status.GetDaemonSets()
```

Comment:
Don't we need an atomic getter for all 3 of them? Otherwise we may get into a race again.

Reply:
We use the same lock for accessing these, so if there is somebody inside the `SetAttributes`, we'll wait until the function is finished and only then read the value.

Reply:
Following can happen:

1. We get DS
2. Somebody sets new DS and Dep
3. We get Dep

Then we don't have DS and Dep from the same batch.

Reply:
Ah I see!

Reply:
Updated.